Sake/Mixer.cs

```diff
@@ -256,62 +256,21 @@ public IEnumerable<ulong> Decompose(IEnumerable<ulong> myInputsParam, IEnumerabl
                 (naiveSet, loss + (ulong)naiveSet.Count * OutputFee)); // The cost is the remaining + output cost.
 
             // Create many decompositions for optimization.
-            var before = DateTimeOffset.UtcNow;
-            do
+            Decomposer.StdDenoms = denoms.Where(x => x <= myInputSum).Select(x => (long)x).ToArray();
+            foreach (var (sum, count, decomp) in Decomposer.Decompose((long)myInputSum, (long)MinAllowedOutputAmountPlusFee, Math.Max(3, naiveSet.Count)))
```

Comment:
With the current code this Decomposer will find the naive solution too so, if we want we can ask it to find only better solutions (excluding the naive one) by:
```suggestion
            foreach (var (sum, count, decomp) in Decomposer.Decompose((long)myInputSum, (long)loss, Math.Max(3, naiveSet.Count)))
```

I strongly recommend this.

Reply:
Yes!